### DISCUSSION TITLE
I can't connect to Vercel Postgresql

### DISCUSSION THREAD

**Discussion by dlorej** [vercel/community#3501] [Help] [answered]
### Summary

When I run this in vercel, i get this error.
`Uncaught (in promise) VercelPostgresError: VercelPostgresError - 'missing_connection_string': You did not supply a 'connectionString' and no 'POSTGRES_URL' env var was found.
`
The error doesn't show up in runtime logs, only in the console.

This is almost my entire code as I am trying to isolate the problem. 

```
import { sql } from "@vercel/postgres";
function App() {
    async function handler() {
        await sql`INSERT INTO testtable VALUES (10,10)`;
    }
    handler();
    return <>Test</>;
}
export default App;
```

I started with create-react-app, i have linked the database  and created the table and can query it from the storage tab on vercel. 
The only problem i can think of is that vercelpostgres doesn't work with react because react needs environment variables to be named starting with REACT_APP_, can anyone help me out and tell me how to fix this problem?

### Steps to Reproduce

`create-react-app`
`npm i @vercel/postgres`
```
async function handler(){
      await sql`CREATE TABLE testtable ( 
         thinga varchar(255),
         thingb varchar(255),
);
```

**Comment by amyegan** [comment, marked as answer]
Hey @dlorej. The first thing to check is that the Postgres environment variables are available in the environment you're using (I'm guessing `development`). Then, if you are running this in your local development environment, make sure you have pulled the environment variables. You can find step-by-step setup instructions at https://vercel.com/docs/storage/vercel-postgres/quickstart#

Your idea about prefixed variables was also right. These variables are not meant for client-side use. You can keep the tokens in server-side code with a Serverless Function. So you might have a file `/api/update-testtable.js` that looks something like this:

```js
import { sql } from '@vercel/postgres';
 
export default async function handler(
  request,
  response,
) {
  try {
    const result =
      await sql`INSERT INTO testtable VALUES (10,10)`;
    return response.status(200).json({ result });
  } catch (error) {
    return response.status(500).json({ error });
  }
}
```

Then `example.com/api/update-testtable` could be called from elsewhere in the code without exposing sensitive values to the client. 

Here are a few docs that cover these topics in more detail:
- [Getting Started with Vercel Postgres](https://vercel.com/docs/storage/vercel-postgres/quickstart#)
- [Vercel Postgres SDK Reference](https://vercel.com/docs/storage/vercel-postgres/sdk#)
- [Serverless Functions Overview](https://vercel.com/docs/concepts/functions/serverless-functions#)

**Reply by dlorej** [reply]
Hello @amyegan, I can't get it to work. I'm new to JavaScript, so I don't know how (request, response) works. I get a type error when I use your code as is. I made some amateur changes to avoid that error but still encounter the same connection string error.

app.js
```
import handler from "./pages/api/cndb";

function App() {
    return (
        <>
            <button onClick={handler}>i am a button</button>
        </>
    );
}
export default App;
```
cndb.js
```
import { sql } from "@vercel/postgres";

export default async function handler(request, response) {
    try {
        const result = await sql`INSERT INTO testtable VALUES (1,1)`;
        console.log("test");
        return response.status(200).json({ result });
    } catch (error) {
        console.log(error);
    }
}
```
This is my file structure.
![image](https://github.com/vercel/community/assets/37584511/5e664adb-74a8-4da2-b47f-c23549c5edc6)

These are my environment variables. I tried adding a connectionString env var but that didnt work.
![image](https://github.com/vercel/community/assets/37584511/c17ba608-0d5b-4f7b-b91b-c5edda376cd4)

Image to show my database connected.
![image](https://github.com/vercel/community/assets/37584511/930dd910-efbf-4a6c-8d74-ff785041bcdb)
